sycl/doc/extensions/fp19Conversion/SYCL_fp19_conversion.asciidoc

```diff
@@ -0,0 +1,328 @@
+----
+
+If a SYCL device has the `ext_fp19_conversion` aspect, then it natively
+supports conversion of values of `float` type to `fp19` and back.
+
+If the device doesn't have the aspect, objects of `fp19` class must not be
+used in the device code.
+
+== New `fp19` class
+
+The `fp19` class below provides the conversion functionality. Conversion
+from `float` to `fp19` is done with rounds to nearest (odd or even), ties away from zero(RNA) rounding
+mode.
+
+[source]
+----
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+class fp19 {
+  using storage_t = uint32_t;
+  storage_t value;
+
+public:
+  fp19() = default;
+  fp19(const fp19 &) = default;
+  ~fp19() = default;
+
+  // Explicit conversion functions
+  static storage_t from_float(const float &a);
+  static float to_float(const storage_t &a);
+
+  // Convert from float to fp19
+  fp19(const float &a);
+  fp19 &operator=(const float &a);
+
+  // Convert from fp19 to float
+  operator float() const;
+
+  // Get fp19 as uint32.
+  operator storage_t() const;
+
+  // Convert to bool type
+  explicit operator bool();
+
+  friend fp19 operator-(fp19 &val) { /* ... */ }
+
+  // OP is: prefix ++, --
+  friend fp19 &operatorOP(fp19 &val) { /* ... */ }
+
+  // OP is: postfix ++, --
+  friend fp19 operatorOP(fp19 &val, int) { /* ... */ }
+
+  // OP is: +=, -=, *=, /=
+  friend fp19 &operatorOP(fp19 &lhs, const fp19 &rhs) { /* ... */ }
+
+  // OP is +, -, *, /
+  friend fp19 operatorOP(const fp19 &lhs, const fp19 &rhs) { /* ... */ }
+  template <typename T>
+  friend fp19 operatorOP(const fp19 &lhs, const T &rhs) { /* ... */ }
+  template <typename T>
+  friend fp19 operatorOP(const T &lhs, const fp19 &rhs) { /* ... */ }
+
+  // OP is ==,!=, <, >, <=, >=
+  friend bool operatorOP(const fp19 &lhs, const fp19 &rhs) { /* ... */ }
+  template <typename T>
+  friend bool operatorOP(const fp19 &lhs, const T &rhs) { /* ... */ }
+  template <typename T>
+  friend bool operatorOP(const T &lhs, const fp19 &rhs) { /* ... */ }
+};
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+----
+
+Table 1. Member functions of `fp19` class.
+|===
+| Member Function | Description
+
+|  `static storage_t from_float(const float &a);`
+|  Explicitly convert from `float` to `fp19`.
+
+|  `static float to_float(const storage_t &a);`
+|  Interpret `a` as `fp19` and explicitly convert it to `float`.
+
+| `fp19(const float& a);`
+| Construct `fp19` from `float`. Converts `float` to `fp19`.
+
+| `fp19 &operator=(const float &a);`
+| Replace the value with `a` converted to `fp19`
+
+| `operator float() const;`
+|  Return `fp19` value converted to `float`.
+
+| `operator storage_t() const;`
+| Return `uint32_t` value, whose bits represent `fp19` value.
+
+| `explicit operator bool() { /* ... */ }`
+| Convert `fp19` to `bool` type. Return `false` if the value equals to
+  zero, return `true` otherwise.
+
+| `friend fp19 operator-(fp19 &val) { /* ... */ }`
```

Comment:
I guess we should also have
```c++
friend fp19 operator+(fp19 &val)`
```
It was also forgotten in the spec up to recently :-) https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_vec_interface